File: frontend/lib/data/dtos/medication_update_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:faims/data/dtos/measurement_unit.dart';

part 'medication_update_dto.g.dart'; 

@JsonSerializable()
class MedicationUpdateDto {
  final String id;
  final String firstAidKitId;
  final String name;
  final int quantity;
  final int minimumQuantity;
  final MeasurementUnit unit;
  final DateTime expirationDate;

  MedicationUpdateDto({
    required this.id,
    required this.firstAidKitId,
    required this.name,
    required this.quantity,
    required this.minimumQuantity,
    required this.unit,
    required this.expirationDate,
  });

  factory MedicationUpdateDto.fromJson(Map<String, dynamic> json) => _$MedicationUpdateDtoFromJson(json);
  Map<String, dynamic> toJson() => _$MedicationUpdateDtoToJson(this);
}

File: Application/DTOs/MedicationUpdateDto.cs
using System.ComponentModel.DataAnnotations;
using Domain;

namespace Application.DTOs;

public record MedicationUpdateDto(
    [Required] Guid Id,
    [Required] Guid FirstAidKitId,
    [Required][StringLength(100, MinimumLength = 2)] string Name,
    [Required] MeasurementUnit Unit,
    [Range(0, int.MaxValue)] int Quantity,
    [Range(1, int.MaxValue)] int MinimumQuantity,
    DateTime ExpirationDate
);